### PR TITLE
docs(configuration): add section about admin password

### PR DESCRIPTION
**Pull request by dedalusj**
Add documentation about new `admin-password` flag introduced by portainer/portainer#752.

### CODE REVIEW

**Comment by deviantony** [comment]
Could you add some documentation about the `--admin-password-path` flag ? Also, it would be nice to add an example of how to run `htpasswd` inside a container (I know it's available in `httpd` for example, e.g. `docker run --rm httpd htpasswd -nb -B admin <PASSWORD>`).

**Comment by dedalusj** [comment]
Done. Let me know if it's not clear enough.

**Comment by dedalusj** [comment]
I updated the docs to reflect the last changes on the Portainer PR.

File: docs/source/configuration.rst [hunk 1]
```diff
@@ -13,6 +13,32 @@ To disable Portainer internal authentication mechanism, start Portainer with the
 
   $ docker run -d -p 9000:9000 -v /var/run/docker.sock:/var/run/docker.sock portainer/portainer --no-auth
 
+Admin password
+==============
+
```
Comment (deviantony) [inline docs/source/configuration.rst]: I think that this section lacks an introduction.

`Portainer allows you to specify an encrypted password from the command line for the admin account. You need to generate the encrypted password first.`

Then, add the section on how to generate an encrypted password first. After that, the section on how to specify the password in the CLI.

File: docs/source/configuration.rst [hunk 2]
```diff
@@ -13,6 +13,32 @@ To disable Portainer internal authentication mechanism, start Portainer with the
...
+To specify the admin password from the command line, start Portainer with the ``--admin-password`` flag:
```
Comment (deviantony) [inline docs/source/configuration.rst]: This is the password from CLI section. Should be written after the password generation section.

File: docs/source/configuration.rst [hunk 3]
```diff
@@ -13,6 +13,32 @@ To disable Portainer internal authentication mechanism, start Portainer with the
...
+
+.. code-block:: bash
+
+  $ docker run -d -p 9000:9000 -v /var/run/docker.sock:/var/run/docker.sock portainer/portainer --admin-password '\$2y\$05\$qFHAlNAH0A.6oCDe1/4W.ueCWC/iTfBMXIHBI97QYfMWlMCJ7N.a6'
+
+The ``\`` are necessary to avoid shell substitution of the ``$`` character inside the container. When running portainer from the binary the admin password can be specified with
```
Comment (deviantony) [inline docs/source/configuration.rst]: Please replace `When running portainer from the binary` with `When running Portainer from the binary (without Docker), the admin password can be specified with:`

File: docs/source/configuration.rst [hunk 4]
```diff
@@ -13,6 +13,32 @@ To disable Portainer internal authentication mechanism, start Portainer with the
...
+
+.. code-block:: bash
+
+  $ portainer --admin-password '$2y$05$qFHAlNAH0A.6oCDe1/4W.ueCWC/iTfBMXIHBI97QYfMWlMCJ7N.a6'
+
+The password hash can be generated with the following command:
```
Comment (deviantony) [inline docs/source/configuration.rst]: Please replace `The password hash can be generated` with `You can generate an encrypted password`

File: docs/source/configuration.rst [hunk 5]
```diff
@@ -95,3 +121,5 @@ The following CLI flags are available:
 * ``--no-auth``: Disable internal authentication mechanism (default: ``false``)
 * ``--external-endpoints``: Enable external endpoint management by specifying the path to a JSON endpoint source in a file
 * ``--sync-interval``: Time interval between two endpoints synchronization requests expressed as a string, e.g. ``30s``, ``5m``, ``1h``... as supported by the `time.ParseDuration method <https://golang.org/pkg/time/#ParseDuration>`_ (default: ``60s``)
+* ``--admin-password``: Admin password in the form ``admin:<hashed_password>``
+* ``--admin-password-path``: Path to a file containing the admin password in the form ``admin:<hashed_password>``
```
Comment (deviantony) [inline docs/source/configuration.rst]: You can remove this line.

**Comment by dedalusj** [comment]
Thanks for the comments. I made the requested changes.

File: docs/source/configuration.rst [hunk 6]
```diff
@@ -13,6 +13,34 @@ To disable Portainer internal authentication mechanism, start Portainer with the
...
+
+.. code-block:: bash
+
+  $ docker run --rm httpd:2.4-alpine htpasswd -nbB admin <password> | cut -d ":" -f 2
+
+To specify the admin password from the command line, start Portainer with the ``--admin-password`` flag:
+
+.. code-block:: bash
+
+  $ docker run -d -p 9000:9000 -v /var/run/docker.sock:/var/run/docker.sock portainer/portainer --admin-password '\$2y\$05\$qFHAlNAH0A.6oCDe1/4W.ueCWC/iTfBMXIHBI97QYfMWlMCJ7N.a6'
```
Comment (deviantony) [inline docs/source/configuration.rst]: I just ran the following:

```
$ docker run --rm -p 9000:9000 -v /var/run/docker.sock:/var/run/docker.sock portainer --admin-password '$2y$05$oLoBoaOT96ODzYn8SWcn0.MxfnOQczVLioOXtQEYKhzhNb3WHArZ6'
```

`$2y$05$oLoBoaOT96ODzYn8SWcn0.MxfnOQczVLioOXtQEYKhzhNb3WHArZ6` is the hash of `coucou`. It's working well without having to escape the **$** char. Is that char protection really needed?

Comment (dedalusj) [inline docs/source/configuration.rst]: No it isn't you are right. The portainer docker image starts from scratch which means no shell substitution takes place. When I tested it originally I was using a ubuntu base image. Sorry about that.

**Pull request merged**